### PR TITLE
[Databases][WIP] Backend server & routes skeleton configuration

### LINKED ISSUE #45: Backend basic setup
**Action item**
Create the backend server logic as well as related routes and controllers modules.

**Deliverables**
Pull request detailing the basic structure of `backend/`.

**Proposed solution**
I'll be using [this tutorial](https://medium.com/crowdbotics/deploy-a-mern-stack-app-on-heroku-b0c255744a70) as a rough guide. 

**Alternatives/WIPs**

### PR DESCRIPTION
**Pull request by wangyyt1013**
## Summary 
This PR introduces the basic backend server configuration, as well as skeleton code for routes and object functions.

## Proposed changes
1. Creation of `/backend` folder in preparation for FE&BE connection in a later PR.
2. Configured `/backend/server.js` to create a basic express server that handles requests to routes specified in the `/backend/routes` folder.
3. Designed basic backend APIs and routes for club profile-related functionalities in `/backend/routes` and `/backend/controllers/`.

## Related issue
Resolves #45 

## Reviewers
@justinbchen @TannerTreasure @ic372 ptal!

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/backend-logic.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+**Action item**
+Summary of the specific task in the sprint that this issue addresses.
+
+**Deliverables**
+As described during the sprint planning meetings.
+
+**Proposed solution**
+Description of specific steps you'll take / changes you'll introduce.
+
+**Alternatives/WIPs**
```
Comment (TannerTreasure) [inline .github/ISSUE_TEMPLATE/backend-logic.md:19]: *Thumbs up*

File: backend/controllers/clubProfileController.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    filterAndSortBy: function(req, res) {
+        // TODO; req.query contains filter and/or sort information
+        // support pagination with req.query 
+        
+    },
+    search: function(req, res) {
+        // TODO; req.query contains search query
+        // support pagination with req.query 
+    }
+}
```
Comment (TannerTreasure) [inline backend/controllers/clubProfileController.js:28]: Understood, Same things we discussed in the meeting last week

**Review by TannerTreasure — COMMENTED** [review]
Seems good to me. I would approve but, you said to use that carefully

**Comment by wangyyt1013** [comment]
Hi yall! As the new PR for the overall backend structure #52 is to be merged into master instead, this PR will serve as a reference/WIP PR on which PR #52 is based on conceptually, and will be closed soon without merging. Still, please feel free to leave any reviews/comments if you do have any; although it's advised to refer to PR #52 for the most updated version of the design. Thanks!

**Pull request closed without merging**